scripts/microshift-cleanup-data.sh

```diff
@@ -0,0 +1,134 @@
+#!/bin/bash
+
+SCRIPT_NAME=$(basename $0)
+# The 'openshift-ovn-kubernetes' namespace must be in the end of the list
+# to allow for pod deletion when MicroShift is stopped
+PODS_NS_LIST=(openshift-service-ca openshift-ingress openshift-dns openshift-storage openshift-ovn-kubernetes)
+FULL_CLEAN=true
+
+function usage() {
+    echo "Stop all MicroShift services, also cleaning their data"
+    echo ""
+    echo "Usage: ${SCRIPT_NAME} <--all | --ovn>"
+    echo "   --all   Clean all MicroShift and OVN data"
+    echo "   --ovn   Clean OVN data only"
+    exit 1
+}
+
+function stop_disable_services() {
+    echo Stopping MicroShift services
+    for service in microshift microshift-etcd ; do
+        systemctl stop --now   ${service} 2>/dev/null
+        systemctl reset-failed ${service} 2>/dev/null
+    done
+
+    if ${FULL_CLEAN} ; then
+        echo Disabling MicroShift services
+        systemctl disable microshift 2>/dev/null
+    fi
+
+    # Killing the processes is the last resort after stopping the services with the 'systemctl stop' command.
+    # Usually, this is only necessary in development environment.
+    pkill -9 --exact microshift
+    pkill -9 --exact microshift-etcd
+}
+
+function stop_clean_pods() {
+    # Wipe the crio data off and return
+    if ${FULL_CLEAN} ; then
+        echo Removing crio container and image storage
+        crio wipe -f &>/dev/null
+        systemctl restart crio
+        return
+    fi
+
+    # It is necessary to stop the pods (OVN-related last) to allow for further termination
+    # of processes (i.e. conmon, etc.) that use the files under /var/run/ovn.
+    # The cleanup of OVN data only works if the files under /var/run/ovn are not in use.
+    echo Removing MicroShift pods
+    for i in ${!PODS_NS_LIST[@]}; do
+        ns=${PODS_NS_LIST[$i]}    
+
+        retries=5
+        while [ ${retries} -gt 0 ] ; do
+            ocp_pods=$(crictl pods --namespace ${ns} -q)
+            if [ "$(echo ${ocp_pods} | wc -w)" -eq 0 ] ; then
+                break
+            fi
+            crictl rmp -f ${ocp_pods} &>/dev/null
+            retries=$(( retries - 1 ))
+        done
+    done
+}
+
+function clean_processes() {
+    if ${FULL_CLEAN} ; then
+        # This operation can only be run in the full clean mode as the removal
+        # of host interfaces is not detected by kubelet and pods cannot be
+        # recovered following the service restart
```

Comment:
@zshi-redhat , can you confirm that this is correct?

Reply:
Yes, it looks good to me.